src/lib/forms/rjsf/widgets/RJSFVocabulary.js

```diff
@@ -34,10 +44,24 @@ export class RJSFVocabulary extends Component {
   };
 
   fetchAll = async () => {
+    const { required } = this.props;
     this.cancellableFetchData = withCancel(this.query());
     try {
+      this.setState({
+        isLoading: true,
+      });
+
       const response = await this.cancellableFetchData.promise;
       const results = response.data.hits.map((hit) => this.serializer(hit));
+      if (!results) {
+        throw Error('No results? This is probably a misconfiguration.');
+      }
+
+      if (!required) {
+        // add an empty element at the beginning so that if it is not required, you can select nothing
+        results.unshift({ key: '', value: undefined, text: '' });
```

Comment:
```suggestion
        results.unshift({ key: '', value: undefined, text: '-' });
```
Comment: Same comment as above (if you agree).